### DISCUSSION TITLE
Add option to set 8-digit code length when manually entering secret keys in Ente Auth

### DISCUSSION THREAD

**Discussion by Pharrcyde** [ente-io/ente#3174] [Enhancements]
I'm using the iOS app version 3.1.3, and I've noticed that when manually entering a secret key for a service that uses 8-digit codes, there is no option to specify the code length. This leads to incorrect code generation for these services.

More details:
1. On iOS (v3.1.3), manually entering secret keys for 8-digit services results in incorrect 6-digit codes.
2. QR code scanning works correctly, as it includes the "digits=8" parameter.
3. Other authenticator apps (e.g., 2FAS) offer the ability to set 8-digit length in advanced settings.

Example QR code data:
otpauth://totp/MyService:MyUsername?secret=MySecretCode&issuer=MyIssuer&algorithm=SHA1&digits=8&period=30

Proposed solution:
Add an option in the manual entry process or in advanced settings to specify the code length (5, 6, 7 or 8 digits) when entering a secret key.

This feature would improve compatibility with services using 8-digit codes.

**Comment by ua741** [comment]
Related https://github.com/ente-io/ente/discussions/1031
